### PR TITLE
Reversing a queue

### LINKED ISSUE #6722: Revesing a Queue
### Description

Reversing a queue using cpp language

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by stutimohta20**
<hr>

## Description 📜

Reversing a queue using cpp language.

Fixes #6722 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [X] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by stutimohta20** [comment]
@Suvraneel .Please review.

File: C_CPP/Data Structures/Queue/readme.md [hunk 1]
```diff
@@ -3,5 +3,5 @@
 - [Queue using linked list](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_LL.cpp)
 - [Queue Operations](Queue-Operations.md)
 - [Implementation of queue using stack](./implementation_of_queue_using_stack.md)
-
+- [Reversing a queue](https://github.com/stutimohta20/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Reversingqueue.md)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/readme.md:6]: Dont you get any hint from the lines above yours?!

Relative link ONLY

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	reverseQueue(q);
+	printQueue(q);
+}
+```
+
+### OUTPUT 
+```
+15 14 13 12 11
+
+```
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:56]: - You've given absolutely NO documentation
- Use syntax highlighting
- Use comments

**Review by Suvraneel — CHANGES_REQUESTED** [review]
After you're done ...
Tag **apeksha235** or **vasu-1**

**Comment by stutimohta20** [comment]
@vasu-1 Please review.

File: C_CPP/Data Structures/Queue/readme.md [hunk 3]
```diff
@@ -3,5 +3,5 @@
 - [Queue using linked list](https://github.com/ishrutik/winter-of-contributing/blob/C_CPP/C_CPP/Data%20Structures/Queue/Queue%20Implementation/Queue_Using_LL.cpp)
 - [Queue Operations](Queue-Operations.md)
 - [Implementation of queue using stack](./implementation_of_queue_using_stack.md)
-
+- [Reversing a queue](C_CPP/Data Structures/Queue/Reversingqueue.md)
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/readme.md:6]:
```suggestion
- [Reversing a queue](./Reversingqueue.md)
```

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	reverseQueue(q);
+	printQueue(q);
+}
+```
+
+### OUTPUT 
+```
+15 14 13 12 11
+
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:70]:
```suggestion
```

Remove unwanted empty lines

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+void printQueue(queue<int>& q)
+{
+	while (!q.empty()) {
+		cout << q.front() << " ";
+		q.pop();
+	}
+}
+
+
+// Main Function
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:51]:
```suggestion
// Main Function
```

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		q.pop();
+	}
+	while (!s.empty()) {
+		q.push(s.top());
+		s.pop();
+	}
+}
+
+
+// Printing the Queue
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:40]:
```suggestion
// Printing the Queue
```

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+1. *Enqueue* - To add elements in the queue.
+2. *Dequeue* - To delete elements from the queue.
+3. *Queue Front* - Gives the front element of the queue.
+4. *Empty* - To check whether the queue is empty or not.
+
+To perform the reversing of a queue ,we first stored it in a stack and then re-inserted it in the queue.
+
+
+**The code for this is given below :**
+```
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:16]:
```suggestion
```cpp
```
For syntax highlighting

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Reversing the Queue 
+
+**QUEUE** - A linear data structure which stores and retrieves data in the **First In First out(FIFO)** order.
+        
+
```
Comment (vasu-1) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:5]:
```suggestion
        
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Do Require Changes

**Comment by stutimohta20** [comment]
@vasu-1 .Made the suggested changes please review once.

**Review by vasu-1 — APPROVED** [review]
LGMT @stutimohta20 

@Suvraneel

**Comment by stutimohta20** [comment]
Thank you :)

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+3. *Queue Front* - Gives the front element of the queue.
+4. *Empty* - To check whether the queue is empty or not.
+
+To perform the reversing of a queue ,we first stored it in a stack and then re-inserted it in the queue.
+
+**The code for this is given below :**
+```cpp
+// Header Files
+
+#include<bits/stdc++.h>
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:16]:
```suggestion

// Header Files
#include<bits/stdc++.h>
```

Function comments just on line above fx

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+using namespace std;
+
+// Function for reversing the Queue
+
+void reverseQueue(queue<int>& q)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:21]:
```suggestion
// Function for reversing the Queue
void reverseQueue(queue<int>& q)
```
Remove extra newlines

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+		q.pop();
+	}
+	while (!s.empty()) {
+		q.push(s.top());
+		s.pop();
+	}
+}
+// Printing the Queue
+
+void printQueue(queue<int>& q)
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:35]:
```suggestion
}

// Printing the Queue
void printQueue(queue<int>& q)
```

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+{
+	while (!q.empty()) {
+		cout << q.front() << " ";
+		q.pop();
+	}
+}
+// Main Function
+
+int main()
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:44]:
```suggestion
}

// Main Function
int main()
```

File: C_CPP/Data Structures/Queue/Reversingqueue.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
+# Reversing the Queue 
+
+**QUEUE** - A linear data structure which stores and retrieves data in the **First In First out(FIFO)** order.	
+*Queue Operations:*
+1. *Enqueue* - To add elements in the queue.
+2. *Dequeue* - To delete elements from the queue.
+3. *Queue Front* - Gives the front element of the queue.
+4. *Empty* - To check whether the queue is empty or not.
+
+To perform the reversing of a queue ,we first stored it in a stack and then re-inserted it in the queue.
```
Comment (Suvraneel) [inline C_CPP/Data Structures/Queue/Reversingqueue.md:10]:
```suggestion
To perform the reversing of a queue ,we first stored it in a stack and then re-inserted it in the queue.
```

* Store the elements
in what order ?

How does this re-insert affect the order
I mean elaborate this one-liner properly..
Ofc its simple but still elaborate as much as you can

**Review by Suvraneel — CHANGES_REQUESTED** [review]

**Comment by stutimohta20** [comment]
Made the changes.

**Comment by Suvraneel** [comment]
@stutimohta20  Excellent ❤️  LGTM 🚀  :octocat:

**Pull request merged**